stable/database/files/nuosm

```diff
@@ -189,74 +371,102 @@ if [ "$restore_source" = ":latest" ]; then
+  trace "trying to reserve first-in"
   $NUOCMD set value --key $first_req --value $HOSTNAME --expected-value ''
   first_in="$( $NUOCMD get value --key $first_req )"
 
-  echo "First-in = $first_in"
+  wotIdid=""
+
+  log "First-in = $first_in"
 
   # if I got in first - perform the restore
   if [ "$first_in" = "$HOSTNAME" ]; then
 
-    echo "I am first-in: $first_in == $HOSTNAME"
+    log "I am first-in: $first_in == $HOSTNAME"
 
-    # [ -z "$restore_source" ] && die -1 "There is no valid LATEST backup - restore of LATEST failed."
+    # [ -z "$restore_source" ] && die 70 "There is no valid LATEST backup - restore of LATEST failed."
 
     # take ownership of the SM startup semaphore
+    trace "Take ownership of SM startup semaphore"
     $NUOCMD set value --key $startup_key/$DB_NAME --value $HOSTNAME --unconditional
 
     # disable all the archive metadata so that get-archive-history will not look for other SMs
+    trace "Disable all archive metadata except my own"
     archive_ids=$( $NUOCMD get archives --db-name $DB_NAME | grep -o "id=[0-9]\+" | grep -o "[0-9]\+")
-    [ -z "$NUODB_DEBUG" ] && echo "archives: $archive_ids"
+    log "archives: $archive_ids"
 
     # delete all archives but my own
     for archv in $archive_ids; do
       [ "$archv" -ne "$myArchive" ] && $NUOCMD delete archive --archive-id $archv
     done
 
     # and restore the data
+    trace "performing restore"
     perform_restore
 
+    wotIdid=""
+
+    log "Restored from $restore_source"
+
     # clear/release shared state
     # $NUOCMD set value --key $first_req --value '' --expected-value $HOSTNAME
     # $NUOCMD set value --key $restore_req --value '' --expected-value $restore_requested
+    log "clearing restore credential request from raft"
     $NUOCMD set value --key $credential_req --value '' --unconditional
 
-
     # any error is a fatal error
-    [ "$retval" = "0" ] || die $retval $error
+    [ $retval -ne 0 ] && die $retval $error
 
   else
     # attempt to restore the same backup that the initial SM is restoring - to reduce SYNC time
+
+    trace "attempting to restore same backup as master SM to secondary SM"
+
     if [ -n "$( echo $restore_source | grep '^[a-z]\+:/[^ ]\+')" -o -d $NUODB_BACKUPDIR/$restore_source ]; then
       perform_restore
 
+      log "Restored secondary archive to match primary restore"
+
       # any error in a SEED restore is logged, but not fatal
-      [ -n "$error" ] && echo "WARNING: Error performing SEED restore: $error"
+      [ -n "$error" ] && log "WARNING: Error performing SEED restore: $error"
     fi
   fi
 
   # wait until it's my turn to startup
+  trace "waiting until it's my turn to start up"
+  retry=0
   until owner=$( $NUOCMD get value --key $startup_key/$DB_NAME ); [ "$owner" = "$HOSTNAME" -o "$NUODB_SEQUENCE_SYNC" = "false" ] ; do
 
+    [ "$owner" = "" -a $retry -gt 0 ] && die $ERR_RESTORE_FAILED "Fatal error in database RESTORE - initial SM has exited with error - $HOSTNAME aborting also"
```

Comment:
What happens if previous SM fails to start after it has been started by `nuodocker`? Possible reasons could be no raft leader, misconfiguration, invalid archive, journal recovery problems, etc. Similar problem is reported in HELM-185.
We will have an owner as `$startup_key/$DB_NAME` is never cleaned but just "passed" between SMs. It is cleaned only in `die` function which is explicitly called if restore scripts fail somewhere.

As discussed before, maybe we should add a timeout while waiting for the last obtained `owner_id` to go into RUNNING. If this timeout, then we can `die` and wait for k8s to restart the pod.